### DISCUSSION TITLE
GrandOrgue and Nektar GX49's

### DISCUSSION THREAD

**Discussion by WombatDave23** [GrandOrgue/grandorgue#1426] [Help]
I have 2 x Nektar GX49 MIDI keyboards plus a home modified 13 key pedal board connected to an Arduino. This setup works under Hauptwerk and plays just fine. I wanted to try Grandorgue but I cannot get both GX49's to talk to Grandorgue. I'm running on an M1 Mac mini with 16GB RAM.

If I monitor the incoming MIDI signals, with MIDI Monitor, I can see all three devices.

I've tried within GrandOrgue to  "listen for MIDI" and "detect complex midi setup" and I have set all three devices to transmit on different channels. The Arduino and the Nektar are all MIDI class compliant devices over USB. Plus I've tried different powered USB hubs all without success.

The detect complex midi setup can find one of the GX49 keyboards and the Arduino but not the second GX49.

I get the same result if all three devices are on MIDI channel 1.

Suggestions would be most welcome.
Thank you

**Comment by willeke1234** [comment]
GrandOrgue identifies keyboards by name instead of ID and two keyboards with the same name doesn't work. Do you see both keyboards in the MIDI Studio window of the Audio MIDI Setup app? It's in the Utilities folder in the Apps folder. Can you change the name of the keyboards?
Note: I don't have any MIDI USB devices so I'm just guessing.

**Reply by William-Purvis** [reply]
I have two GX61's which I am hoping to use with GO, and I suspect I am going to hit the same problem. I'm running under Linux Ubuntu
with Jack. I, too, would appreciate hints as to how to get round this? I have set up the keyboards individually to separate channels, but the above suggests this is not enough! There's no way to modify the  keyboard name, as such.

**Reply by William-Purvis** [reply]
Just ran a quick test with the two GX61's to my laptop. Qjackctl shows the devices as:
  Impact GX61 [20] (capture): Impact GX61 MIDI 1
  Impact GX61 [20] (capture): Impact GX61 MIDI 2
And both appear as above in the GO Midi Settings window under Device pull-down, but no response from the last one plugged in. Unplugging one and the remaining one will work, but then restoring the unplugged one is not responsive.
How do other people manage? I can't believe everyone uses 2 or three totally different keyboards?

**Comment by WombatDave23** [comment]
I’ve had the same issue. I can’t find a way forward. On 13 Apr 2023, at 16:55, William-Purvis ***@***.***> wrote:﻿
Just ran a quick test with the two GX61's to my laptop. Qjackctl shows the devices as:
Impact GX61 [20] (capture): Impact GX61 MIDI 1
Impact GX61 [20] (capture): Impact GX61 MIDI 2
And both appear as above in the GO Midi Settings window under Device pull-down, but no response from the last one plugged in. Unplugging one and the remaining one will work, but then restoring the unplugged one is not responsive.
How do other people manage? I can't believe everyone uses 2 or three totally different keyboards?

**Reply by William-Purvis** [reply]
OK, making a little progress. The Nektar keyboards each have two midi ports - one is the 'normal' port, and the second (MIDI 2) is for
interaction with DAWs. I don't use a DAW, and have not tried to chase up the protocols for that.

I noticed that the device in the setting window include a number:
jack: a2j:Impact GX61 [20] (capture).... and
jack: a2j:Impact GX61 [24] (capture)....

For some reason I seem to have acquired three versions of MIDI 1, two with [20] and one with [24].
By experiment, selecting one of each number seems to work.

Only problem now is that for some reason GO is refusing to produce any sound! I think the duplicate [20] is something left in ALSA when I unplugged and re-plugged the keyboard? Got to sort all this out for the church organ, where the only interaction with the PC is the power switch!
Anyone who really understands all this, please help!

**Comment by davidgritter** [comment]
I just tried to plug in two identical devices, in this case raspberry pi pico's with the identical midi program and midi device name.  One appears in qjackctl as  20:PICO Organ pedals and the second as 24:PICO Organ Pedals.   When I start GrandOrgue, which had no previous knowledge of these midi interfaces, it automatically made connections to both of them.  From your comments above, where your devices showed up in QjackCtl with slightly different naming convention that you may be using an older version of Jack or possibly of ALSA.  Newer version deal with multiple of the same midi or sound card devices much better.

When using multiple devices, the first one connected will always get the lower number.  This can be dealt with in ALSA configuration if they have different Vid's or Pids. (https://alsa.opensrc.org/MultipleCards) However the best way to deal with this in GrandOrgue for midi devices is to configure your keyboards to use 9x Note from "any device" and have each device send on different midi channels.  Normally if you connect a new midi device to GrandOrgue and don't get sound, its because the keyboard hasn't been configured for "any device'"

When sending MIDI to a device  (to turn LED's on and off for example) GrandOrgue seems to want you to specify a single device for each message. and the choice is only one entry for multiple devices.  In that case you are better off not using identical devices with current versions of alsa  (some previous versions of alsa gave identical devices unique names with a long hexadecimal number appended which depended on which USB port the device was plugged in to, but latest alsa doesn't do that anymore, as far as I can tell)

**Comment by midimusic** [comment]
It is well known to programmers that two usb midi interfaces with the same name are not recognised by the operating system as being different. The operating system presents both interfaces as having the same name. However GO (via RtMidi) makes up a name so that, to the user, the two interfaces APPEAR to have different names.

This causes no end of confusion because the user does not know what has happened and thinks both interfaces have been correctly recognised. This is a long standing problem which is not likely to go away soon. The best solution is NEVER connect two identical midi interfaces to a computer via usb as it will always cause unexpected problems.

csw900

**Reply by rousseldenis** [reply]
A good solution is to use a midi merger interface to do so.
